site/blog/2022/nodejs-deploy-codepipeline.mdx

```diff
@@ -0,0 +1,221 @@
+
+## The NodeJS application
+
+In this example, we are going to build and deploy a very simple NodeJS application, that simply will output **Hello World!** to the browser. The application will be served to port 80 by Nginx, that will be acting as a reverse proxy for the node app. The application is hosted on a public github repository: https://github.com/iasql/iasql-code-example-node
+
+The indications for building and deploying the whole application are defined in the `appspec.yml` file: https://github.com/iasql/iasql-code-example-node/blob/main/appspec.yml . This file contains all the information needed by CodeDeploy to manage the whole lifecycle of the application (Complete reference can be seen at https://docs.aws.amazon.com/codedeploy/latest/userguide/reference-appspec-file-example.html#appspec-file-example-server).
+
+This sample configuration is divided in two steps: **files** and **hooks**. 
+**Files** section allows to specify where to copy the contents of the application. The contents of this application are divided between main content of the app, and specific nginx configuration.
+**Hooks** section allows to define scripts to tackle every step of the app's lifecycle. In this case those steps are covered:
+- BeforeInstall: script to install all needed dependencies to run the application. It is used to install all packages and dependencies
+- AfterInstalll: script triggered after all the dependencies have been installed but before the app is receiving any traffic. It is the best place to perform additional installation steps of the application, such as `npm install`
+- ApplicationStart: script that is actually managing the start of the application, with command like `npm start`
+
+For this specific example the deployment of the application will be performed into an EC2 instance, and content will just be exposed via the **public ip**. Additional steps such as using Autoscaling groups, or Load Balancers would improve the availability of the application, but are not covered on this post.
+
+## Pre-requisites
+
+The source code and configuration of this sample is using GitHub on IaSQL organization account. In order to run the example, please fork the https://github.com/iasql/iasql-code-example-node repository under an account you can control, as it needs some additional permissions to connect with AWS Codepipeline.
+
+Next step is to grant AWS permissions to the forked repo. One way of doing that is by using personal tokens. Please follow this documentation to create the token: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token. Please ensure that the token has **repo** and **repo_hook** permissions. Copy the generated token, as it will be used later when configuring the pipeline.
```

Comment:
```suggestion
Next step is to grant AWS permissions to the forked repo. One way of doing that is by using [personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). Please follow this documentation to create the token and ensure that the token has **repo** and **repo_hook** permissions. Copy the generated token, as it will be used later when configuring the pipeline.
```